src/components/portfolio/Portfolio.js

```diff
@@ -22,7 +25,108 @@ class Portfolio extends Component {
     )
   };
 
+  constructor(props) {
+    super(props);
+    this.state = {
+      stockDetails: [],
+      cryptoDetails: [],
+      isLoading: true,
+      showStart: false
+    };
+    this.onDelete = this.onDelete.bind(this);
+  }
+
+  onDelete(index) {
+    let stateObj = this.state;
```

Comment:
Try not to mutate the state this way. Instead just use `setState` to mutate it. i.e this code will look like:
```
   const stockDetails = this.state.stockDetails.filter(
     (item, idx) => idx !== selectedIndex
    );

   this.setState({
       ...this.state,
       stockDetails
   });

   const portfolioDetails = {
      stockDetails
   };
```